applications/luci-app-wireguard/luasrc/view/wireguard.htm

```diff
@@ -0,0 +1,280 @@
+<%#
+ Copyright 2016-2017 Dan Luedtke <mail@danrl.com>
+ Licensed to the public under the Apache License 2.0.
+-%>
+
+<%
+  function strip(s)
+    return (s:gsub("^%s*(.-)%s*$", "%1"))
+  end
+
+  function is_valid(s)
+    return (string.len(strip(s)) > 0)
+  end
+
+  function peer_add(peers, public_key)
+    table.insert(peers, {
```

Comment:
It might be a bit easier to implement this as a hash table, where the public key is the key and the rest of the items are a dictionary of values.